src/map.cpp

```diff
@@ -618,6 +618,60 @@ void map::move_vehicle( vehicle &veh, const tripoint &dp, const tileray &facing
             veh.skidding = true;
             veh.turn( coll_turn );
         }
+        // accept new position
+        // if submap changed, we need to process grid from the beginning.
+        // TODO: Allow vehicles to get displaced vertically
+        const tripoint &veh_start=veh.global_pos3();
```

Comment:
BTW you could move all those functions to `vehicle` class, like you did with `operate_planter`. You could give them `dp` as an argument if you need it there.

Or even better: have a single vehicle class `on_move` for all the functions that trigger on movement. This could trigger plows, reapers, scoops etc.